staging/src/k8s.io/component-helpers/scheduling/corev1/helpers.go

```diff
@@ -30,3 +36,52 @@ func PodPriority(pod *v1.Pod) int32 {
 	// name of the pod was empty. So, we resolve to the static default priority.
 	return 0
 }
+
+// IsExtendedResourceName returns true if:
+// 1. the resource name is not in the default namespace;
+// 2. resource name does not have "requests." prefix,
+// to avoid confusion with the convention in quota
+// 3. it satisfies the rules in IsQualifiedName() after converted into quota resource name
+func IsExtendedResourceName(name v1.ResourceName) bool {
+	if IsNativeResource(name) || strings.HasPrefix(string(name), v1.DefaultResourceRequestsPrefix) {
+		return false
+	}
+	// Ensure it satisfies the rules in IsQualifiedName() after converted into quota resource name
+	nameForQuota := fmt.Sprintf("%s%s", v1.DefaultResourceRequestsPrefix, string(name))
```

Comment:
why not `+` instead of `Sprintf`. I assume it should be faster

Reply:
> why not + instead of Sprintf.

"old skool" code. 

> I assume it should be faster

Concatenation is usually the slowest. [bytes.Buffer](https://hermanschaaf.com/efficient-string-concatenation-in-go/) could be better though it depends on how many times `IsExtendedResourceName` is actually invoked.

Reply:
I wouldn't use concatenation in a loop, that's definitely quadratic. But for 2 strings, it should be as fast as using a Buffer (or even faster). `Sprintf` would be slower than both.

Reply:
https://gist.github.com/dtjm/c6ebc86abe7515c988ec

concat is good here